rct229/rules/section5/section5rule24.py

```diff
@@ -0,0 +1,259 @@
+            return {
+                **data,
+                # TODO this function will likely need to be revised to RMD level later.
+                "scc_dict_b": get_surface_conditioning_category_dict(
+                    climate, building_b
+                ),
+                "target_u_factor_mix": target_u_factor_mix,
+                "target_u_factor_res": target_u_factor_res,
+                "target_u_factor_nonres": target_u_factor_nonres,
+                "target_u_factor_semiheated": target_u_factor_semiheated,
+            }
+
+        def list_filter(self, context_item, data=None):
+            surface_b = context_item.baseline
+            scc_dict_b = data["scc_dict_b"]
+            return (get_opaque_surface_type(surface_b) == OST.ABOVE_GRADE_WALL) and (
+                scc_dict_b[surface_b["id"]] != SCC.UNREGULATED
+            )
+
+        class AboveGradeWallRule(RuleDefinitionListIndexedBase):
+            def __init__(self):
+                super(Section5Rule24.BuildingRule.AboveGradeWallRule, self).__init__(
+                    rmrs_used=UserBaselineProposedVals(False, True, False),
+                    each_rule=Section5Rule24.BuildingRule.AboveGradeWallRule.SubsurfaceRule(),
+                    index_rmr="baseline",
+                    list_path="subsurfaces[*]",
+                    required_fields={
+                        "$.subsurfaces[*]": [
+                            "classification",
+                            "glazed_area",
+                            "opaque_area",
+                            "u_factor",
+                        ]
+                    },
+                )
+
+            def create_data(self, context, data=None):
+                surface_b = context.baseline
+                scc_dict_b = data["scc_dict_b"]
+                return {**data, "scc": scc_dict_b[surface_b["id"]]}
+
+            def list_filter(self, context_item, data=None):
+                subsurface_b = context_item.baseline
+                return (
+                    subsurface_b is None
+                    or subsurface_b["classification"] != DOOR
+                    or subsurface_b["glazed_area"] > subsurface_b["opaque_area"]
+                )
+
+            class SubsurfaceRule(RuleDefinitionBase):
+                def __init__(self):
+                    super(
+                        Section5Rule24.BuildingRule.AboveGradeWallRule.SubsurfaceRule,
+                        self,
+                    ).__init__(
+                        rmrs_used=UserBaselineProposedVals(False, True, False),
+                    )
+
+                def get_calc_vals(self, context, data=None):
+                    subsurface_b = context.baseline
+                    return {
+                        "subsurface_u_factor": subsurface_b["u_factor"],
+                    }
+
+                def rule_check(self, context, calc_vals=None, data=None):
+                    scc = data["scc"]
+                    target_u_factor = 0.0
```

Comment:
Ii might slightly prefer to initialize this to None so that we'll get an exception if we somehow get to line 259 without truly setting this value.